CMakeLists.txt

```diff
@@ -185,10 +185,8 @@ if(OPENMP_ENABLE)
   #message(STATUS "****** finding OpenMP support")
   FIND_PACKAGE(OpenMP)
   if(OPENMP_FOUND)
-    add_definitions(-DHAVE_OPENMP)
-	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CDD_FLAGS} ${OpenMP_CXX_FLAGS}")
-    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+    add_compile_options($<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}> $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)
+    link_libraries(-fopenmp)
```

Comment:
IS this going to work with MSVC?  seems like it is the gcc option

Reply:
I think the better way of doing this is a check if CMAKE>3.9 and if so use the target otherwise we will need to set the flags